src/Peachpie.Runtime/Attributes.cs

```diff
@@ -287,6 +287,26 @@ public sealed class NotNullAttribute : Attribute
 
     }
 
+    /// <summary>
+    /// Compiler generated attribute denoting that the method returns a specific type
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public sealed class ReturnTypeAttribute : Attribute
+    {
+        /// <summary>
+        /// The type containing exported functions.
+        /// </summary>
+        public string SyntaxType { get; private set; }
+
+        public bool AllowsNull { get; private set; }
```

Comment:
We already have attribute `[return: NotNull]` which should be emitted if the return type cannot be `null`.